src/ssl/support.cc

```diff
@@ -559,7 +559,11 @@ Ssl::VerifyCallbackParameters::At(Security::Connection &sconn)
 }
 
 // "dup" function for SSL_get_ex_new_index("cert_err_check")
-#if SQUID_USE_CONST_CRYPTO_EX_DATA_DUP
+#if OPENSSL_VERSION_MAJOR >= 3
+static int
+ssl_dupAclChecklist(CRYPTO_EX_DATA *, const CRYPTO_EX_DATA *, void **,
+                    int, long, void *)
+#elif SQUID_USE_CONST_CRYPTO_EX_DATA_DUP
```

Comment:
AFAICT, this change implies that SQUID_USE_CONST_CRYPTO_EX_DATA_DUP-setting code needs to be updated (or, better, the whole thing should be refactored to move all this mess to ./configure code). Otherwise, falsy SQUID_USE_CONST_CRYPTO_EX_DATA_DUP in the PR code incorrectly implies that SSL_get_new_ex_index() dup callback accepts non-const `CRYPTO_EX_DATA *`. I do not insist on these changes though.

Reply:
This change is about parameter 3 (```void *```) becoming a ```void **`` and parameter 2 being cemented as the ```const CRYPTO_EX_DATA``` for all OpenSSL 3+.

Older OpenSSL versions require non-const ```CRYPTO_EX_DATA```. Except some release had backported patches of partial v3 API - adding only the ```const CRYPTO_EX_DATA```. This latter group is detected by SQUID_USE_CONST_CRYPTO_EX_DATA_DUP.